bin/reassure-tests.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+set -e
+
+# Move out of bin directory to Gutenberg project root
+cd ../
+
+BASELINE_BRANCH=${BASELINE_BRANCH:="trunk"}
+
+# Required for `git switch` on CI
+git fetch origin
+
+# Gather baseline perf measurements
+git switch "$BASELINE_BRANCH"
+npm --cwd ci --force
```

Comment:
> I imagine the usage of the --force flag would increase the duration of this CI task. Is there a particular reason to use it in this file?

The `--force` flag comes from following Reassure's [example CI script](https://github.com/callstack/reassure/blob/e564320cc93310206e8477e688b98fc2f40a3aff/examples/native/reassure-tests.sh#L12-L13), with the explanation that "this line is required because Reassure packages are imported from this monorepo and might require rebuilding." I'm not certain if that is a direct reference to the `--force` flag specifically, or something else. I'm also not certain if `--force` is actually required. 

> I wonder if we could reuse the project's setup-node task like the mobile unit tests action. Possibly not, since it appears we need to run npm ci twice on different branches.
This could be useful to reduce the amount of time the tests run, and would align with the rest of our Github actions. I'm not certain if it would be possible, either, but naturally I would be up for any change that makes running the performance tests more performant. :) 

I'll remove the CI code from this PR to another branch. On the project board, I'll note revisiting both of these items when we start the CI task. I'll leave this thread Unresolved for now in case you have any other thoughts to add between now and then.